### PR TITLE
feature(SpringCloud): add Java 17 support #15814

### PR DESCRIPTION
**Pull request by rtrompier**

### CODE REVIEW

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @rtrompier - Thanks for opening this PR.  Sadly we can't make direct changes to vendor'd files as they'd simply be overwritten when Go manages it's dependencies. As a first-time contributor, GHA doesn't run automatically, which contains a test to ensure these files are intact and unmodified as Go needs them to be, so I'm afraid it would fail this and not be mergeable.

I've checked the API specs and this change is available in a later version of the preview API, however, so we can look to upgrade that to enable your change to add Java17. If this is not something you're comfortable doing, just say, and one of the team can try to take a look as soon as time allows.

**Comment by rtrompier** [comment]
Hi @jackofallops 
Yes, really sorry, it's my first contribution and first time with GO language ;)
I didn't see I modify the vendor source ... Maybe the folder should be ignored by git ?
If someone more experimented can have a look about this request it will be really great.

**Comment by jackofallops** [comment]
Hi @rtrompier - No apologies necessary! We appreciate all efforts to help improve the provider 😃 

The vendor'd files are (currently) included to provide the best consistent build experience, so cannot be excluded via `.gitignore` for now, this may change in future. As a first time contributor the GitHub Actions require a maintainer to approve to run, since I knew it would fail I've not done this on this occasion. fwiw - any file under the `vendor` folder should not be modified directly as Go manages those via [tidy](https://go.dev/ref/mod#go-mod-tidy) and [vendor](https://go.dev/ref/mod#go-mod-vendor).

I'll mention this to the team and see if anyone has capacity to check if the API version can be bumped with no breaking changes, which will unblock this and allow you to update this PR on top of the newer API. Does that work for you?

**Comment by rtrompier** [comment]
Perfect ! 
I wait for your feedback ;) 
Have a good day.

File: internal/services/springcloud/spring_cloud_java_deployment_resource.go
Comment (katbyte) [inline internal/services/springcloud/spring_cloud_java_deployment_resource.go:447]: This doesn't appear to exist in the SDK?

**Review by katbyte — CHANGES_REQUESTED** [review]
Hey @rtrompier - Thanks for the PR - it looks like that constant doesn't exit in the currently vendored in SDK? perhapes do we need to upgrade to a new SDK version to enable this property?

**Comment by tombuildsstuff** [comment]
hey @rtrompier 

As mentioned above unfortunately this change needs to be added [to the upstream Swagger Definition where the Azure SDK is generated from](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/appplatform/resource-manager/Microsoft.AppPlatform/stable/2022-04-01) - as such whilst we'd like to thank you for this contribution I'm going to close this PR for the moment as this change needs to first be made in the Swagger before this is usable in the Azure SDK (and ultimately in Terraform).

Thanks!

**Pull request closed without merging**
